app/controllers/tickets_controller.rb

```diff
@@ -46,4 +48,8 @@ def ticket_update_params
   def load_ticket
     @ticket = Ticket.find(params[:id])
   end
+
+  def validate_ticket
+    redirect_to root_path, alert: '降車済みの切符です。' if @ticket.exited_gate_id
```

Comment:
```suggestion
    redirect_to root_path, alert: '降車済みの切符です。' if @ticket.exited_gate.present?
```

idの有無を調べるより、モデルの有無を調べる方がオブジェクト指向らしくて良いと思います。